### PR TITLE
make optimizer and count_num_param can handle params directly.

### PR DESCRIPTION
**Pull request by siaimes**
make optimizer and count_num_param can handle params directly.

Signed-off-by: siaimes <34199488+siaimes@users.noreply.github.com>

### CODE REVIEW

File: dassl/optim/optimizer.py [hunk 1]
```diff
@@ -36,50 +37,50 @@ def build_optimizer(model, optim_cfg):
...
-            )
-
-        if isinstance(model, nn.DataParallel):
-            model = model.module
-
-        if isinstance(new_layers, str):
-            if new_layers is None:
-                warnings.warn(
-                    "new_layers is empty, therefore, staged_lr is useless"
+    if param_groups is None:
```
Comment (KaiyangZhou) [inline dassl/optim/optimizer.py:40]: Looks like `param_group` and `staged_lr` cannot be used at the same time.

Here we need to throw an error or warning that once `param_group` is given, `staged_lr` shouldn't be True.

File: dassl/utils/torchtools.py [hunk 2]
```diff
@@ -251,16 +251,25 @@ def open_specified_layers(model, open_layers):
...
 
     Args:
         model (nn.Module): network model.
-
+        params: network model`s params.
     Examples::
         >>> model_size = count_num_param(model)
     """
-    return sum(p.numel() for p in model.parameters())
+    if model:
```
Comment (KaiyangZhou) [inline dassl/utils/torchtools.py:263]: Use `if model is not None`, which is more informative.

Also add space between blocks to make the code more readable.

File: dassl/optim/optimizer.py (hunk 1 above)
Comment (siaimes) [reply dassl/optim/optimizer.py:40]: Yes, at this time, the learning rate needs to be attached to the param_groups by the user.

**Comment by KaiyangZhou** [comment]
Thank you @siaimes for the updates.

Btw, please run `linter.sh` and fix the inconsistencies next time when submitting a PR.
